c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/operation/DebugOperationHandler.java

```diff
@@ -0,0 +1,257 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
+import org.apache.nifi.c2.client.api.C2Client;
+import org.apache.nifi.c2.protocol.api.C2Operation;
+import org.apache.nifi.c2.protocol.api.C2OperationAck;
+import org.apache.nifi.c2.protocol.api.C2OperationState;
+import org.apache.nifi.c2.protocol.api.C2OperationState.OperationState;
+import org.apache.nifi.c2.protocol.api.OperandType;
+import org.apache.nifi.c2.protocol.api.OperationType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DebugOperationHandler implements C2OperationHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DebugOperationHandler.class);
+
+    private static final String C2_CALLBACK_URL_NOT_FOUND = "C2 Server callback URL was not found in request";
+    private static final String SUCCESSFUL_UPLOAD = "Debug bundle was uploaded successfully";
+    private static final String UNABLE_TO_CREATE_BUNDLE = "Unable to create debug bundle";
+
+    static final String TARGET_ARG = "target";
+    static final String NEW_LINE = "\n";
+
+    private final C2Client c2Client;
+    private final String configDir;
+    private final String logDir;
+    private final Predicate<String> contentFilter;
+
+    private DebugOperationHandler(C2Client c2Client, String configDir, String logDir, Predicate<String> contentFilter) {
+        this.c2Client = c2Client;
+        this.configDir = configDir;
+        this.logDir = logDir;
+        this.contentFilter = contentFilter;
+    }
+
+    public static DebugOperationHandler create(C2Client c2Client, String configDir, String logDir, Predicate<String> contentFilter) {
+        if (c2Client == null) {
+            throw new IllegalArgumentException("C2Client should not be null");
+        }
+        if (isBlank(configDir)) {
+            throw new IllegalArgumentException("configDir should not be not null or empty");
+        }
+        if (isBlank(logDir)) {
+            throw new IllegalArgumentException("logDir should not be not null or empty");
+        }
+        if (contentFilter == null) {
+            throw new IllegalArgumentException("Exclude sensitive filter should not be null");
+        }
+
+        return new DebugOperationHandler(c2Client, configDir, logDir, contentFilter);
+    }
+
+    @Override
+    public OperationType getOperationType() {
+        return TRANSFER;
+    }
+
+    @Override
+    public OperandType getOperandType() {
+        return DEBUG;
+    }
+
+    @Override
+    public C2OperationAck handle(C2Operation operation) {
+        Function<C2OperationState, C2OperationAck> operationAckCreate = operationAck(operation);
```

Comment:
Is there a specific need for this function based approach? Extending the function to get 2 parameters looks to be sufficient and simpler. What do you think? Or maybe there is something that I'm missing?

Reply:
Tried to use a more functional approach, but agree, Java syntax is a bit cluncky for this. Refactored to a 2 parameters accepting method